From 74a6bbf4dc822f9a1ce27da140a7eac7e0211523 Mon Sep 17 00:00:00 2001
From: X512 <danger_mail@list.ru>
Date: Tue, 22 Jun 2021 12:30:15 +0900
Subject: ahci: remove packed attribute from register structure

It cause per-byte access instead of 32 bit on GCC/riscv64 that breaks operation.

All fields are 32 bit so alignment is already fine and packing is not needed.

Change-Id: Ie96eac6615c9326e84608be1c667bc5d3600c508
---
 src/add-ons/kernel/busses/scsi/ahci/ahci_defs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/add-ons/kernel/busses/scsi/ahci/ahci_defs.h b/src/add-ons/kernel/busses/scsi/ahci/ahci_defs.h
index 578defa00b..b016dab232 100644
--- a/src/add-ons/kernel/busses/scsi/ahci/ahci_defs.h
+++ b/src/add-ons/kernel/busses/scsi/ahci/ahci_defs.h
@@ -129,7 +129,7 @@ typedef struct {
 	uint32		devslp;			// Device Sleep
 	uint32		res[10];		// Reserved
 	uint32		vendor[4];		// Vendor Specific
-} _PACKED ahci_port;
+} ahci_port;
 
 
 enum {
-- 
2.30.2

